File: Algorithms/data-structure/selection_sort.py
from typing import MutableSequence


def selection_sort(a: MutableSequence) -> None:
    n = len(a)
    for i in range(n - 1):
        min = i
        for j in range(i + 1, n):
            if a[j] < a[min]:
                min = j
        print(a)
        a[i], a[min] = a[min], a[i]
        print(a)

    return a


selection_sort([7, 5, 9, 0, 3, 1, 6, 2, 4, 8])

File: inflearn/javascript/selection-sort.js
function selectionSort(arr) {
  let answer = [];

  for (let i = 0; i < arr.length; i++) {
    let min = Math.min(arr);
    answer.push(min);
    arr.splice(min);
  }
  console.log(answer);

  return answer;
}

selectionSort([13, 5, 11, 7, 23, 15]);
